### PR TITLE
feat(author-jinja): load jinja extensions from plugins

### PR DESCRIPTION
**Pull request by rahearn**
## Types of changes

- [ ] Hot fix (emergency fix and release)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation (change which affects the documentation site)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [x] My code follows the code style of this project.
- [x] Documentation for my change is up to date?
- [x] My PR meets testing requirements.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Summary

Builds on the proof of concept at #1684 and:

* moves the methods in the previously new `ssp_interface` into a set of jinja filters. Example switch in use:

`{% for party in ssp_interface.get_parties_for_role(ssp.metadata.responsible_parties, "prepared-by") %}` becomes
`{% for party in ssp.metadata.responsible_parties | parties_for_role("prepared-by", ssp) %}`

`{{ ssp_interface.safe_retrieval(address, 'addr_lines', []) | join(' ') }}` becomes `{{ address.addr_lines | as_list | join(' ') }}`

* refactors the jinja code into a module to better organize it and avoid some circular dependencies
* enables loading extensions from plugins, searching under: `trestle_{plugin_name}.jinja_ext`
* updates plugin docs

## Key links:

- #1719 
- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar. If you are working for a fork a maintainer will reach out, if required.

### CODE REVIEW

**Comment by butler54** [comment]
Hi @rahearn  - sorry for the delay. I'll look at this in the next few days :).

File: tests/trestle/core/jinja/filters_test.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# -*- mode:python; coding:utf-8 -*-
+
+# Copyright (c) 2020 IBM Corp. All rights reserved.
```
Comment (jpower432) [inline tests/trestle/core/jinja/filters_test.py:3]:
```suggestion
# Copyright (c) 2024 The OSCAL Compass Authors.
```

**Review by jpower432 — CHANGES_REQUESTED** [review]
@rahearn Thanks for this submission! Wanted to add some early feedback here. Requesting new source code files (the newly created `jinja` package in this case) have the updated copyright info. I added a suggestion on one of the files to what this should be updated to.

File: trestle/core/plugins.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+_discovered_plugins = {
+    name: importlib.import_module(name)
+    for finder,
+    name,
+    ispkg in pkgutil.iter_modules()
+    if name.startswith('trestle_')
+}
+
+
+def discovered_plugins(search_module: str) -> Iterator[Tuple[str, Any]]:
```
Comment (butler54) [inline trestle/core/plugins.py:36]: This is a nice improvement :)

File: trestle/core/jinja/base.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+"""Trestle core.jinja base class."""
+from jinja2 import lexer, nodes
+from jinja2.environment import Environment
+from jinja2.ext import Extension
+
+
+class TrestleJinjaExtension(Extension):
+    """Class to define common methods to be inherited from for use in trestle."""
+
+    # This
```
Comment (butler54) [inline trestle/core/jinja/base.py:25]: Think `#this` needs to be cleaned up.

**Review by butler54 — APPROVED** [review]
LGTM - This is really great work.  @jpower432's picked up one comment otherwise the only thing that i've seen here which needs to be tidied here is a dangling comment.

**Review by jpower432 — APPROVED** [review]
LGTM. Thanks for the contribution @rahearn!

**Comment by jpower432** [comment]
@rahearn Your branch needs to be up to date with the base branch and then it should be good to merge.

**Pull request merged**